src/main/webapp/app/exercises/programming/manage/services/programming-exercise-simulation.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from 'app/app.constants';
+import { EntityResponseType, ProgrammingExerciseService } from 'app/exercises/programming/manage/services/programming-exercise.service';
+import { ProgrammingExercise } from 'app/entities/programming-exercise.model';
+import { HttpClient } from '@angular/common/http';
+import { map } from 'rxjs/operators';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable({ providedIn: 'root' })
+export class ProgrammingExerciseSimulationService {
```

Comment:
clearly document that this is only intended for local testing and should never be used in the production environment

Reply:
added comments to highlight that this particular functionality is only for testing purposes